include/grub/zfs/vdev_impl.h

```diff
@@ -23,34 +23,33 @@
 #ifndef _SYS_VDEV_IMPL_H
 #define	_SYS_VDEV_IMPL_H
 
-#define	VDEV_SKIP_SIZE		(8 << 10)
-#define	VDEV_BOOT_HEADER_SIZE	(8 << 10)
+#define	VDEV_PAD_SIZE		(8 << 10)
 #define	VDEV_PHYS_SIZE		(112 << 10)
 #define	VDEV_UBERBLOCK_RING	(128 << 10)
 
-/* ZFS boot block */
-#define	VDEV_BOOT_MAGIC		0x2f5b007b10cULL
-#define	VDEV_BOOT_VERSION	1		/* version number	*/
-
-typedef struct vdev_boot_header {
-	grub_uint64_t	vb_magic;		/* VDEV_BOOT_MAGIC	*/
-	grub_uint64_t	vb_version;		/* VDEV_BOOT_VERSION	*/
-	grub_uint64_t	vb_offset;		/* start offset	(bytes) */
-	grub_uint64_t	vb_size;		/* size (bytes)		*/
-	char		vb_pad[VDEV_BOOT_HEADER_SIZE - 4 * sizeof (grub_uint64_t)];
-} vdev_boot_header_t;
-
 typedef struct vdev_phys {
 	char		vp_nvlist[VDEV_PHYS_SIZE - sizeof (zio_eck_t)];
 	zio_eck_t	vp_zbt;
 } vdev_phys_t;
 
+typedef enum vbe_vers {
+	VB_RAW,
+	VB_NVLIST
+} vb_evers_t;
```

Comment:
This struct is primarily used by the kernel when writing the contents of the boot envblock in vdev_label_write_bootenv.  I'm not sure I understand the second half of the question; "also" what? That's the only thing we're setting with the values in this enum.

Reply:
Sorry that was a typo. I meant that `vdev_boot_envblock_t->vbe_version` is set to the value of this enum.

Given that the kernel is the primary user of the enum, I'm assuming that it will be copied into the zfs code base. In this case, why do we keep it here? Can you elaborate on how those versions will be used? For instance what does "VB_NVLIST" imply? Also, if this is something that defines on-disk format, should we assign explicit integer values to the each enum name?

Reply:
The kernel is the primary setter of the enum, but we check the value here to make sure that the version in use is supported by GRUB. I can add a comment expanding on the meanings of the values, and assign them specific integers.